application/views/catalog/partials/footer.php

```diff
@@ -19,18 +19,18 @@
 
 	var sub_category;
 
-	var advanced_search = <?= (int)$advanced_search ?>;
+	var advanced_search = <?= (int)(isset($advanced_search) ?: 0) ?>;
 
 	var primary_key = <?= (int)$primary_key ?>;
 
-	var search_page = <?= (int)($search_page ?: 1) ?>;
+	var search_page = <?= (int)(isset($search_page) ?: 1) ?>;
 	set_advanced_form_page(search_page);
 
 	var search_order = 'alpha';
 
 	var project_type = 'either';
 
-	var q = <?= json_encode($search_value ?: '', JSON_HEX_TAG | JSON_HEX_QUOT) ?>;
+	var q = <?= json_encode(isset($search_value) ?: '', JSON_HEX_TAG | JSON_HEX_QUOT) ?>;
```

Comment:
I think this should be:
`var q = <?= json_encode(isset($search_value) ? $search_value : '', JSON_HEX_TAG | JSON_HEX_QUOT) ?>;`

Making that change seems to fix the search from the home page always searching for "true" (see my recent comments in #106).

I expect the other two changes in this commit should be updated similarly.

I do wonder if this is even the best way to fix this at all or if code somewhere else should be changed to always set these variables. I don't understand the overflow well enough to say one way or the other right now.

Reply:
> I think this should be:
> `var q = <?= json_encode(isset($search_value) ? $search_value : '', JSON_HEX_TAG | JSON_HEX_QUOT) ?>;`


Yep, good catch. As it is, using the shorthand ```?:``` means the expression evaluates to the value of the first operand if the latter is True-ish, and because we've started using ```isset``` that's just ```True```